src/proxy/momento/src/metrics/util.rs

```diff
@@ -0,0 +1,25 @@
+use goodmetrics::{GaugeDimensions, GaugeFactory, HistogramHandle, SumHandle};
+
+pub fn proxy_sum_gauge(g: &GaugeFactory, name: &'static str) -> SumHandle {
+    g.dimensioned_gauge_sum("momento_proxy", name, Default::default())
+}
+
+pub fn proxy_request_ok_gauge(g: &GaugeFactory, rpc: &'static str) -> SumHandle {
+    g.dimensioned_gauge_sum("momento_proxy", "Ok", GaugeDimensions::new([("rpc", rpc)]))
+}
+
+pub fn proxy_request_error_gauge(g: &GaugeFactory, rpc: &'static str) -> SumHandle {
+    g.dimensioned_gauge_sum(
+        "momento_proxy",
+        "Error",
+        GaugeDimensions::new([("rpc", rpc)]),
+    )
+}
+
+pub fn proxy_request_latency_gauge(g: &GaugeFactory, rpc: &'static str) -> HistogramHandle {
+    g.dimensioned_gauge_histogram(
+        "momento_proxy",
+        "latency",
+        GaugeDimensions::new([("rpc", rpc)]),
```

Comment:
You have the option to remove the other two counters and only use this histogram, with rpc and result dimensions. This would help people see the latency of successful requests, which often present different latency from failed requests.

Reply:
Nice. That sounds more compact.

To do this I would need to instantiate the `HistogramHandle` in order to start the `TimerGuard`, but I wouldn't know the value of `result` at that time. Can I populate that dimension later?

Reply:
Updated in [23976b9](https://github.com/pelikan-io/pelikan/pull/149/commits/23976b93cb9b91c0e8ad7ea37aa07a159efb8be7)